src/test/java/org/opentripplanner/standalone/config/ExampleConfigTest.java

```diff
@@ -0,0 +1,57 @@
+package org.opentripplanner.standalone.config;
+
+import static java.util.Map.entry;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.function.Consumer;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.opentripplanner.standalone.config.framework.JsonSupport;
+import org.opentripplanner.standalone.config.framework.json.NodeAdapter;
+import org.opentripplanner.standalone.config.framework.project.EnvironmentVariableReplacer;
+import org.opentripplanner.test.support.FilePatternSource;
+
+public class ExampleConfigTest {
+
+  Map<String, String> env = Map.ofEntries(
+    entry("OTP_GCS_WORK_DIR", "/var/entur"),
+    entry("OTP_GCS_BASE_GRAPH_PATH", "/var/entur"),
+    entry("OTP_GCS_BUCKET", "/var/entur")
+  );
+
+  @FilePatternSource(pattern = "docs/examples/**/router-config.json")
+  @ParameterizedTest(name = "Check validity of {0}")
+  void routerConfig(Path filename) {
+    testConfig(filename, a -> new RouterConfig(a, true));
+  }
+
+  @FilePatternSource(
+    pattern = { "docs/examples/**/build-config.json", "test/performance/**/build-config.json" }
+  )
+  @ParameterizedTest(name = "Check validity of {0}")
+  void buildConfig(Path filename) {
+    testConfig(filename, a -> new BuildConfig(a, true));
+  }
+
+  private void testConfig(Path path, Consumer<NodeAdapter> buildConfig) {
+    try {
+      var json = Files.readString(path);
+      var replaced = EnvironmentVariableReplacer.insertEnvironmentVariables(json, json, env);
+      var node = JsonSupport.jsonNodeFromString(replaced);
+      var a = new NodeAdapter(node, getClass().getSimpleName());
```

Comment:
Should we use `ConfigFileLoader` to do thIs?

Reply:
I would have to make the following modifications to ConfigFileLoader:
- make it work with Paths/Files, because right now it operates on folders
- make the env vars lookup injectable just like the EnvironmentVariableReplacer

Reply:
Do you think that is worth it?

Reply:
I think it is ok, we can take a new look at this after the rework around configs is done